### PR TITLE
refactor(core): change file system handling in CLI to account for NgTree

### PR DESCRIPTION
**Pull request by ViktorSlavov**

### CODE REVIEW

File: packages/ng-schematics/src/component/index_spec.ts [hunk 1]
```diff
@@ -23,8 +24,6 @@ describe("component",  () => {
 		const tree = runner.runSchematicAsync("component",
 			{ name: "my-combo", template: "combo", templateInst: mockInst, skipRoute: false }, Tree.empty());
 		tree.subscribe(state => {
-			expect(mockInst.generateConfig).toHaveBeenCalledWith("my-combo", {});
-			expect(mockInst.registerInProject).toHaveBeenCalledWith("", "my-combo", { skipRoute: false, modulePath: undefined });
```
Comment (damyanpetev) [inline packages/ng-schematics/src/component/index_spec.ts]: Erm, shouldn't `registerInProject` still be called?

File: packages/core/util/App.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	public static initialize() {
+		this.container.set(FS_TOKEN, new FsFileSystem());
+		this.container.set(FS_TYPE_TOKEN, FsTypes.physical);
+	}
+
+	public static get workDir() {
+		return this.container.get<FsTypes>(FS_TYPE_TOKEN) ? process.cwd() : "";
+	}
+
+	public static setVirtual(fileSystem: IFileSystem) {
```
[the remark this answers is not in the document]
Comment (ViktorSlavov) [reply packages/core/util/App.ts]: Thanks, I've moved it as a helper in schematics/utils

File: packages/core/util/Container.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+export enum FsTypes {
+	virtual = 0,
+	physical = 1
+}
+
+export const FS_TYPE_TOKEN: string = "FS_TYPE";
```
[the remark this answers is not in the document]
Comment (ViktorSlavov) [reply packages/core/util/Container.ts]: Yup, thanks for pointing it out!

File: packages/ng-schematics/src/cli-config/index.ts [hunk 4]
```diff
@@ -67,6 +68,7 @@ function importBrowserAnimations(): Rule {
 
 function importStyles(): Rule {
 	return (tree: Tree) => {
+		App.setVirtual(new NgTreeFileSystem(tree));
```
[the remark this answers is not in the document]
Comment (ViktorSlavov) [reply packages/ng-schematics/src/cli-config/index.ts]: Move it up the chain

File: packages/ng-schematics/src/component/index.ts [hunk 5]
```diff
@@ -65,14 +66,20 @@ export function singleComponent(templateOptions: TemplateOptions, skipRoute: boo
...
 }
 export function component(options: ComponentOptions): Rule {
-	return async (_tree: Tree, context: SchematicContext) => {
-
+	return async (tree: Tree, context: SchematicContext) => {
+		App.initialize();
 		const addedComponents: TemplateOptions[] = [];
 		const templateManager = new SchematicsTemplateManager();
 		const projLib = templateManager.getProjectLibrary("angular", "igx-ts");
+		const properties = {
```
[the remark this answers is not in the document]
Comment (ViktorSlavov) [reply packages/ng-schematics/src/component/index.ts]: "Magic, do not touch" :D

I've added a comment why this (and a similar one in `ng-new` are needed)

File: packages/ng-schematics/src/component/index_spec.ts (hunk 1 above)
Comment (ViktorSlavov) [reply packages/ng-schematics/src/component/index_spec.ts]: Yeah, you're correct, this was actually failing for a reason - I've reverted the change here are adjusted the schematics code

**Comment by damyanpetev** [comment]
@ViktorSlavov oh.. just noticed the target branch 🙄 Re-targeting to vNext

File: packages/ng-schematics/src/utils/cli-config.ts [hunk 6]
```diff
@@ -41,17 +41,10 @@ function GetCliConfig(tree: Tree): Config {
 function getPort(workspace: WorkspaceSchema) {
 	const targetProjectName = workspace.defaultProject;
 	const projectServe = targetProjectName
-		? workspace.projects[targetProjectName].architect.serve.options
+		? (workspace.projects[targetProjectName].architect || { serve: { }}).serve.options
```
Comment (damyanpetev) [inline packages/ng-schematics/src/utils/cli-config.ts]: Erm.. wut

Comment (ViktorSlavov) [reply packages/ng-schematics/src/utils/cli-config.ts]: Typescript was going crazy on my end, I've reverted this

**Pull request merged**